File: app/src/main/java/com/example/ui/SettingsScreen.kt
package com.example.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.core.Palette

@Composable
fun SettingsScreen(
    bgmEnabled: Boolean,
    sfxEnabled: Boolean,
    voiceEnabled: Boolean,
    useBanglaDigits: Boolean,
    showFpsOverlay: Boolean,
    onBgmToggled: (Boolean) -> Unit,
    onSfxToggled: (Boolean) -> Unit,
    onVoiceToggled: (Boolean) -> Unit,
    onBanglaDigitsToggled: (Boolean) -> Unit,
    onFpsOverlayToggled: (Boolean) -> Unit,
    onBackClicked: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Palette.voidBlack)
            .padding(20.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "⚙ SETTINGS",
            color = Palette.gold,
            fontSize = 24.sp,
            fontWeight = FontWeight.Black,
            fontFamily = FontFamily.Monospace,
            modifier = Modifier.padding(vertical = 16.dp)
        )

        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            SettingRow(
                label = "Chiptune BGM",
                value = bgmEnabled,
                onCheckedChange = onBgmToggled,
                testTag = "bgm_toggle"
            )
            SettingRow(
                label = "Retro SFX",
                value = sfxEnabled,
                onCheckedChange = onSfxToggled,
                testTag = "sfx_toggle"
            )
            SettingRow(
                label = "Bangla Voice Lines",
                value = voiceEnabled,
                onCheckedChange = onVoiceToggled,
                testTag = "voice_toggle"
            )
            SettingRow(
                label = "Bangla Digits (০১২৩)",
                value = useBanglaDigits,
                onCheckedChange = onBanglaDigitsToggled,
                testTag = "digits_toggle"
            )
            SettingRow(
                label = "Debug FPS & Pool Overlay",
                value = showFpsOverlay,
                onCheckedChange = onFpsOverlayToggled,
                testTag = "fps_toggle"
            )
        }

        RetroArcadeButton(
            text = "⬅  B A C K",
            bgColor = Palette.slate,
            textColor = Palette.paper,
            testTag = "back_button",
            onClick = onBackClicked
        )
    }
}

@Composable
private fun SettingRow(
    label: String,
    value: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    testTag: String
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(Palette.shadow, RoundedCornerShape(8.dp))
            .border(1.dp, Palette.slate, RoundedCornerShape(8.dp))
            .padding(14.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = label,
                color = Palette.paper,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = FontFamily.Monospace
            )
            Switch(
                checked = value,
                onCheckedChange = onCheckedChange,
                modifier = Modifier.testTag(testTag),
                colors = SwitchDefaults.colors(
                    checkedThumbColor = Palette.gold,
                    checkedTrackColor = Palette.leaf,
                    uncheckedThumbColor = Palette.ash,
                    uncheckedTrackColor = Palette.ink
                )
            )
        }
    }
}

File: flutter_app/lib/ui/settings_screen.dart
import 'package:flutter/material.dart';
import '../core/palette.dart';

class SettingsScreen extends StatelessWidget {
  final bool bgmEnabled;
  final bool sfxEnabled;
  final bool voiceEnabled;
  final bool useBanglaDigits;
  final bool showFpsOverlay;
  final ValueChanged<bool> onBgmToggled;
  final ValueChanged<bool> onSfxToggled;
  final ValueChanged<bool> onVoiceToggled;
  final ValueChanged<bool> onBanglaDigitsToggled;
  final ValueChanged<bool> onFpsOverlayToggled;
  final VoidCallback onBackClicked;

  const SettingsScreen({
    required this.bgmEnabled,
    required this.sfxEnabled,
    required this.voiceEnabled,
    required this.useBanglaDigits,
    required this.showFpsOverlay,
    required this.onBgmToggled,
    required this.onSfxToggled,
    required this.onVoiceToggled,
    required this.onBanglaDigitsToggled,
    required this.onFpsOverlayToggled,
    required this.onBackClicked,
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Palette.voidBlack,
      padding: const EdgeInsets.all(20),
      child: Column(
        children: [
          const Text("SETTINGS", style: TextStyle(color: Palette.gold, fontSize: 24, fontFamily: 'monospace', fontWeight: FontWeight.w900)),
          const SizedBox(height: 16),
          Expanded(
            child: Column(
              children: [
                _buildSettingRow("Chiptune BGM", bgmEnabled, onBgmToggled),
                _buildSettingRow("Retro SFX", sfxEnabled, onSfxToggled),
                _buildSettingRow("Bangla Voice Lines", voiceEnabled, onVoiceToggled),
                _buildSettingRow("Bangla Digits (0-9)", useBanglaDigits, onBanglaDigitsToggled),
                _buildSettingRow("Debug FPS & Pool Overlay", showFpsOverlay, onFpsOverlayToggled),
              ],
            ),
          ),
          const SizedBox(height: 16),
          _buildButton("BACK", Palette.slate, Palette.paper, onBackClicked),
        ],
      ),
    );
  }

  Widget _buildSettingRow(String label, bool value, ValueChanged<bool> onChanged) {
    return Container(
      width: double.infinity,
      margin: const EdgeInsets.symmetric(vertical: 6),
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color: Palette.shadow,
        borderRadius: BorderRadius.circular(8),
        border: Border.all(color: Palette.slate, width: 1),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(label, style: const TextStyle(color: Palette.paper, fontSize: 15, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
          Switch(
            value: value,
            onChanged: onChanged,
            activeThumbColor: Palette.gold,
            activeTrackColor: Palette.leaf,
            inactiveThumbColor: Palette.ash,
            inactiveTrackColor: Palette.ink,
          ),
        ],
      ),
    );
  }

  Widget _buildButton(String text, Color bg, Color textColor, VoidCallback onTap) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        width: double.infinity,
        height: 50,
        decoration: BoxDecoration(
          color: bg,
          borderRadius: BorderRadius.circular(6),
          border: Border.all(color: Palette.ink, width: 2),
        ),
        child: Center(
          child: Text(text, style: TextStyle(color: textColor, fontSize: 16, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
        ),
      ),
    );
  }
}
